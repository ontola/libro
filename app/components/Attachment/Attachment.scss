@import '../shared/config';

.Attachment {
  border: $grey-border;
  border-radius: $border-radius-base;
  display: flex;
  max-width: 20em;
  overflow: hidden;
  position: relative;
}

.Attachment__primary {
  align-items: center;
  display: flex;
  flex-grow: 1;
  flex-shrink: 1;
  overflow: hidden;
  padding-left: .4em;
  position: relative;
}

.Attachment__icon {
  color: color('grey', 'light');
  position: relative;
  z-index: 1;
}

.Attachment__text {
  color: color('grey', 'mid-dark');
  margin: 0 .6em;
  overflow: hidden;
  position: relative;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.Attachment__inside-button {
  align-items: center;
  border-left: $grey-border;
  display: flex;
  flex-grow: 0;
  flex-shrink: 0;
  height: 1.7em;
  justify-content: center;
  margin-right: auto;
  position: relative;
  width: 1.7em;
  z-index: 1;

  &:hover {
    background-color: color('grey', 'xx-light');

    .Attachment__icon {
      color: color('grey', 'xx-light-foreground-small');
    }
  }
}
