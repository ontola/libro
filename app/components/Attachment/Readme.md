Attachments show a file that can be opened, downloaded and previewed.

## Not downloaded

    <Attachment  
      title="Some_document.pdf"
      url="http://downloads.palass.org/annual_meeting/2000/confabs2000.pdf"
      isDownloaded={false}
      isDownloading={false}
      percentageDownloaded={0}
      hasPreview={true}
    />

## Downloading

    <Attachment  
      title="Some_document.pdf"
      url="http://downloads.palass.org/annual_meeting/2000/confabs2000.pdf"
      isDownloaded={false}
      isDownloading={true}
      percentageDownloaded={78}
      hasPreview={true}
    />

## Downloaded

    <Attachment  
      title="Some_document.pdf"
      url="http://downloads.palass.org/annual_meeting/2000/confabs2000.pdf"
      isDownloaded={true}
      percentageDownloaded={100}
      hasPreview={true}
    />

## No preview possible

    <Attachment  
      title="Some_document.pdf"
      url="http://downloads.palass.org/annual_meeting/2000/confabs2000.pdf"
      hasPreview={false}
    />

## Huge filename / title

    <Attachment  
      title="Having_a_huge_filename_is_actually_quite_common_in_governmental_organisations.pdf"
      url="http://downloads.palass.org/annual_meeting/2000/confabs2000.pdf"
    />
