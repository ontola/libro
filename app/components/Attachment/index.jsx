import PropTypes from 'prop-types';
import React from 'react';
import FontAwesome from 'react-fontawesome';

import { byteToSize } from '../../helpers/numbers';
import LDLink from '../LDLink';

import './Attachment.scss';

const propTypes = {
  /** Link to the file */
  contentUrl: PropTypes.string.isRequired,
  fileFormat: PropTypes.string,
  /** Filesize in bytes */
  fileSize: PropTypes.string,
  /** Label that is displayed */
  title: PropTypes.string.isRequired,
};

const previewableFileFormats = [
  // 'application/pdf',
];

const Attachment = ({
  title,
  fileFormat,
  fileSize,
  contentUrl,
}) => (
  <div className="Attachment">
    <LDLink
      className="Attachment__primary"
      title={title}
    >
      <FontAwesome className="Attachment__icon" name="file" />
      <div className="Attachment__text" data-test="Attachment-title">{title}</div>
    </LDLink>
    <a
      download
      className="Attachment__inside-button"
      data-test="Attachment-download"
      href={contentUrl}
      rel="noopener noreferrer"
      target="_blank"
      title={`Download ${byteToSize(fileSize)}`}
    >
      <FontAwesome className="Attachment__icon" name="download" />
    </a>
    {previewableFileFormats.includes(fileFormat)
      && (
      <a
        className="Attachment__inside-button"
        data-test="Attachment-preview"
        href="/"
        title="Bekijk voorbeeldweergave"
      >
        <FontAwesome className="Attachment__icon" name="caret-right" />
      </a>
      )
    }
  </div>
);

Attachment.propTypes = propTypes;

export default Attachment;
