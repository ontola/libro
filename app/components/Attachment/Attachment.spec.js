import Attachment from './index';

argUnit(Attachment, () => {
  setProp('url', () => 'http://argu.co/');
  setProp('title', () => 'Attachment label');

  it('renders', () => {
    expect(subject).toMatchSnapshot();
  });

  it('renders the label', () => {
    expect(subject.find(marker('title'))).toHaveText('Attachment label');
  });
});
