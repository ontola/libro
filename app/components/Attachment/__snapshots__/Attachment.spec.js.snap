// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Attachment renders 1`] = `
<div
  className="Attachment"
>
  <withLinkCtxWrapper
    className="Attachment__primary"
    title="Attachment label"
  >
    <FontAwesome
      className="Attachment__icon"
      name="file"
    />
    <div
      className="Attachment__text"
      data-test="Attachment-title"
    >
      Attachment label
    </div>
  </withLinkCtxWrapper>
  <a
    className="Attachment__inside-button"
    data-test="Attachment-download"
    download={true}
    rel="noopener noreferrer"
    target="_blank"
    title="Download NaN undefined"
  >
    <FontAwesome
      className="Attachment__icon"
      name="download"
    />
  </a>
</div>
`;
